e2e/cluster.go

```diff
@@ -71,7 +71,7 @@ func ClusterKubenet(ctx context.Context, t *testing.T) (*Cluster, error) {
 
 func ClusterKubenetAirgap(ctx context.Context, t *testing.T) (*Cluster, error) {
 	clusterKubenetAirgapOnce.Do(func() {
-		cluster, err := prepareCluster(ctx, t, getKubenetClusterModel("abe2e-kubenet-airgap"), true)
+		cluster, err := prepareCluster(ctx, t, getKubenetClusterModel("abe2e-airgap"), true)
```

Comment:
Are you sure we want every e2e test running with the network plugin? I just want to make sure we are running the e2e's as close as possible to what a customer would be running without assuming they have anything installed

Since I'm assuming Test_Ubuntu2204Gen2ContainerdAirgapped_K8sNotCached needs the plugin to pass, maybe just that test should use the AzureNetwork cluster?

Reply:
in fact we do not support kubenet cluster for network isolated cluster. So that it should be azure network cluster. However, my latest test fails with node not ready. Still restricting the test range. Maybe in this pr we may remain kubenet and unblock the test first

Reply:
I just ran `Test_Ubuntu2204Gen2ContainerdAirgapped_K8sNotCached` on local (with a test rg and acr) and it was able to pass. Are you still seeing node not ready?

Reply:
For what its worth, the kubelet logs show 
```
Nov 26 09:14:11 abe2e-2024-11-26-bj4f-ubuntu2204gen2containerdairgappedk8000000 kubelet[3491]: I1126 09:14:11.829439    3491 csi_plugin.go:99] kubernetes.io/csi: Trying to validate a new CSI Driver with name: disk.csi.azure.com endpoint: /var/lib/kubelet/plugins/disk.csi.azure.com/csi.sock versions: 1.0.0
Nov 26 09:14:11 abe2e-2024-11-26-bj4f-ubuntu2204gen2containerdairgappedk8000000 kubelet[3491]: I1126 09:14:11.829463    3491 csi_plugin.go:112] kubernetes.io/csi: Register new plugin with name: disk.csi.azure.com at endpoint: /var/lib/kubelet/plugins/disk.csi.azure.com/csi.sock
Nov 26 09:15:32 abe2e-2024-11-26-bj4f-ubuntu2204gen2containerdairgappedk8000000 kubelet[3491]: E1126 09:15:32.347559    3491 kubelet_node_status.go:456] "Node not becoming ready in time after startup"
Nov 26 09:15:32 abe2e-2024-11-26-bj4f-ubuntu2204gen2containerdairgappedk8000000 kubelet[3491]: E1126 09:15:32.839128    3491 kubelet.go:2892] "Container runtime network not ready" networkReady="NetworkReady=false reason:NetworkPluginNotReady message:Network plugin returns error: cni plugin not initialized"
```
for why node is not ready when using getAzureNetworkClusterModel